### PR TITLE
OLS-201: Support for HEAD HTTP methods for liveness endpoints

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Support for HEAD HTTP methods for liveness endpoints

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-201](https://issues.redhat.com//browse/OLS-201)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

## Testing
- CI

### CODE REVIEW

File: ols/app/endpoints/health.py [hunk 1]
```diff
@@ -8,12 +13,22 @@
...
-def liveness_probe() -> dict[str, str]:
+def liveness_probe_get_method() -> dict[str, str]:
     """Live status of service."""
     return {"status": "1"}
+
+
+@router.head("/readiness")
+def readiness_probe_head_method() -> None:
+    """Ready status of service."""
+
```
Comment (onmete) [inline ols/app/endpoints/health.py:30]: It would be better to return None. This might confuse someone.

Comment (tisnik) [reply ols/app/endpoints/health.py:31]: yup

**Pull request merged**